pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -100,6 +102,34 @@ func (r *MigPlan) GetDestinationCluster(client k8sclient.Client) (*MigCluster, e
 	return GetCluster(client, r.Spec.DestMigClusterRef)
 }
 
+// GetSourceIdentity - Get the corresponding user for a given source identity
+// token
+// Returns "" if the srcToken doesn't authenticate
+func (r *MigPlan) GetSourceIdentity(client k8sclient.Client) (string, error) {
+	secret, err := r.GetIdentitySecret(client)
+	if err != nil {
+		return "", err
+	}
+	tokenReview := v1beta1.TokenReview{
+		Spec: v1beta1.TokenReviewSpec{
+			Token: string(secret.Data["srcToken"]),
+		},
+	}
+	err = client.Create(context.TODO(), &tokenReview)
+	if err != nil {
+		return "", err
+	}
+	if !tokenReview.Status.Authenticated {
+		return "", errors.New("token did not authenticate with source cluster")
```

Comment:
Small suggestion: Given the scope of the nature of the method, seems `token not authenticated` would be more concise and sufficient.